Extension/src/LanguageServer/configurations.ts

```diff
@@ -739,29 +741,47 @@ export class CppProperties {
         const result: string[] = [];
         if (paths) {
             paths = this.resolveDefaults(paths, defaultValue);
-            paths.forEach(entry => {
-                const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
-                result.push(...entries);
-            });
-        }
-        if (!glob) {
-            return result;
+            if (paths && !glob) {
+                paths.forEach(entry => {
+                    const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
+                    result.push(...entries);
+                });
+                return result;
+            }
         }
 
         const globResult: string[] = [];
-        for (let res of result) {
-            // fastGlob will expand the ending double wildcard. temporary strip them before expanding
-            const recursive: boolean = res.endsWith('**');
-            if (recursive) {
-                res = res.slice(0, res.length - 2);
-            }
-            // fastGlob can't deal with backslash-separated path => remove them
-            const normalized: string = res.replace(/\\/g, '/');
-            const cwd: string = this.rootUri?.fsPath?.replace(/\\/g, '/') || '';
-            // fastGlob silently strip non-found paths. limit that behavior to dynamic paths only
-            const matches: string[] = fastGlob.isDynamicPattern(normalized) ?
-                fastGlob.sync(normalized, { onlyDirectories: true, cwd}) : [res];
-            globResult.push(...matches.map(s => recursive ? s + '**' : s));
+        if (paths) {
+            for (let res of paths) {
+                let counter: integer = 0;
+                let replacementString: string = '';
+                const lastIndex: integer = res.length - 1;
+                // Detect and glob all wildcard variations by looking at last character in the path first.
+                for (let i: integer = lastIndex; i >= 0; i--) {
+                    if (res[i] === '*') {
+                        counter++;
+                        replacementString = replacementString.replace (/^/, res[i]);
+                    } else if (res[i] === '/' || res[i] === '\\') {
+                        replacementString = replacementString.replace (/^/, res[i]);
+                        counter++;
+                        break;
+                    } else {
+                        break;
+                    }
+                }
+                res = res.slice(0, res.length - counter);
```

Comment:
should it be `res.length - counter - 1`?

Reply:
No, this works fine because we already index with an offset

Reply:
where is the offset?  `counter` is initialized with `0`.

Reply:
the counter already counts the appropriate number of characters, I can demo it to you

Reply:
Sure. I was thinking the second parameter was an 'index', not a 'length' and an index is one less than the length to that index.